Tests/LeafMarkdownTests/LeafTests.swift

```diff
@@ -1,73 +1,85 @@
+            let syntax = try parser.parse()
+            let unresolved = UnresolvedDocument(name: "markdowntest", raw: syntax)
+            let resolver = ExtendResolver(document: unresolved, dependencies: [])
+            template = try resolver.resolve(rootDirectory: config.rootDirectory)
+        } catch {
+            XCTFail()
+        }
     }
     
     // MARK: - Tests
 
     func testRunTag() throws {
         let inputMarkdown = "# This is a test\n\nWe have some text in a tag"
-        let data = TemplateData.dictionary(["data": .string(inputMarkdown)])
         let expectedHtml = "<h1>This is a test</h1>\n<p>We have some text in a tag</p>\n"
 
-        let result = try renderer.render(template: template.data(using: .utf8)!, data).wait()
-        let resultString = String(data: result.data, encoding: .utf8)!
+        let result = try renderer.render(template, context: ["data": .string(inputMarkdown)])
+        let resultString = result.getString(at: 0, length: result.readableBytes)
         XCTAssertEqual(resultString, expectedHtml)
     }
 
     func testNilParameterDoesNotCrashLeaf() throws {
-        let data = TemplateData.dictionary(["data": .null])
         let expectedHtml = ""
-
-        let result = try renderer.render(template: template.data(using: .utf8)!, data).wait()
-        let resultString = String(data: result.data, encoding: .utf8)!
+        let result = try renderer.render(template, context: ["data": .null])
+        let resultString = result.getString(at: 0, length: result.readableBytes)
         XCTAssertEqual(resultString, expectedHtml)
     }
 
     func testStripHtml() throws {
         let inputMarkdown = "<br>"
-        let data = TemplateData.dictionary(["data": .string(inputMarkdown)])
         let expectedHtml = "<!-- raw HTML omitted -->\n"
 
-        let result = try renderer.render(template: template.data(using: .utf8)!, data).wait()
-        let resultString = String(data: result.data, encoding: .utf8)!
+        let result = try renderer.render(template, context: ["data": .string(inputMarkdown)])
+        let resultString = result.getString(at: 0, length: result.readableBytes)
         XCTAssertEqual(resultString, expectedHtml)
 
     }
 
     func testDoNotStripHtml() throws {
-
-        let queue = EmbeddedEventLoop()
-        let container = BasicContainer(config: .init(), environment: .testing, services: .init(), on: queue)
-        let tag = Markdown(options: [])
-        var leafTagConfig = LeafTagConfig.default()
-        leafTagConfig.use(tag, as: tag.name)
-        let renderer = LeafRenderer(config: LeafConfig(tags: leafTagConfig, viewsDir: "", shouldCache: false),
-                                     using: container)
+        let loop = EmbeddedEventLoop()
+        let config = LeafConfiguration(rootDirectory: Process().currentDirectoryPath)
+        var tags = defaultTags
+        tags["markdown"] = Markdown(options: [.init(rawValue: 1 << 17)])
```

Comment:
This can now be updated with `.unsafe` if you set the minimum SwiftMarkdown version to 0.6.1 as that's now been added